controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitRequestHandler.java

```diff
@@ -87,29 +86,38 @@ public CommitRequestHandler(final StreamMetadataStore streamMetadataStore,
     public CompletableFuture<Void> execute(CommitEvent event) {
         String scope = event.getScope();
         String stream = event.getStream();
-        int epoch = event.getEpoch();
         OperationContext context = streamMetadataStore.createContext(scope, stream);
-        log.debug("Attempting to commit available transactions on epoch {} on stream {}/{}", event.getEpoch(), event.getScope(), event.getStream());
+        log.debug("Attempting to commit available transactions on stream {}/{}", event.getScope(), event.getStream());
 
         CompletableFuture<Void> future = new CompletableFuture<>();
 
-        tryCommitTransactions(scope, stream, epoch, context)
+        // Note: we will ignore the epoch in the event. It has been deprecated. 
+        // The logic now finds the smallest epoch with transactions and commits them.
+        tryCommitTransactions(scope, stream, context)
                 .whenComplete((r, e) -> {
                     if (e != null) {
                         Throwable cause = Exceptions.unwrap(e);
                         // for operation not allowed, we will report the event
                         if (cause instanceof StoreException.OperationNotAllowedException) {
-                            log.debug("Cannot commit transaction on epoch {} on stream {}/{}. Postponing", epoch, scope, stream);
+                            log.debug("Cannot commit transaction on stream {}/{}. Postponing", scope, stream);
                         } else {
-                            log.error("Exception while attempting to commit transaction on epoch {} on stream {}/{}", epoch, scope, stream, e);
+                            log.error("Exception while attempting to commit transaction on stream {}/{}", scope, stream, e);
                         }
                         future.completeExceptionally(cause);
                     } else {
-                        log.debug("Successfully committed transactions on epoch {} on stream {}/{}", epoch, scope, stream);
+                        if (r >= 0) {
+                            log.debug("Successfully committed transactions on epoch {} on stream {}/{}", r, scope, stream);
+                        } else {
+                            log.debug("No transactions found in committing state on stream {}/{}", r, scope, stream);
+                        }
                         if (processedEvents != null) {
-                            processedEvents.offer(event);
+                            try {
+                                processedEvents.offer(event);
+                            } catch (Exception ex) {
+                                // ignore, this processed events is only added for enabling unit testing this class
```

Comment:
What's `processedEvents`, is that state of the object only used for testing?

Reply:
Yes, its only for testing. This was present in CommitEventProcessor and used in its tests. Maybe as a different issue we can clean this up and write our tests differently so as to not include test specific fields in the class.

Reply:
If it needs fixing or improving in a separate PR, please create an issue to record it.

Reply:
https://github.com/pravega/pravega/issues/3566